core/src/main/java/org/elasticsearch/plugins/PluginInfo.java

```diff
@@ -95,6 +95,12 @@ public static PluginInfo readFromProperties(Path dir) throws IOException {
         if (version == null) {
             throw new IllegalArgumentException("Property [version] is missing for plugin [" + name + "]");
         }
+        String pluginName = props.getProperty("name");
```

Comment:
Will do, I was not going deeper into the temp dir thing because this would change a lot in the plugin manager and I was told it will get a real update sooner or later anyway.